compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -2012,7 +2013,7 @@ impl<'tcx> Operand<'tcx> {
         Operand::Constant(box Constant {
             span,
             user_ty: None,
-            literal: ty::Const::zero_sized(tcx, ty),
+            literal: ConstantSource::Ty(ty::Const::zero_sized(tcx, ty)),
```

Comment:
why are you using `ConstantSource::Ty` instead of `Val` here?

do you prefer `ty::Const` values over allocations where possible?

Reply:
Mostly in order to change as little as possible. `ConstantSource::Val` is essentially dead code in this PR